### PR TITLE
Add new native method and refactoring method names

### PR DESCRIPTION
**Pull request by wggihan**
Add  a new native method to check the next record availability. Rename existing readTextRecord native method to nextTextRecord while keeping the current functionality.
This resolves https://github.com/ballerina-lang/ballerina/issues/4530

### CODE REVIEW

File: modules/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/HasNextTextRecord.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import org.ballerinalang.natives.AbstractNativeFunction;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.Receiver;
+import org.ballerinalang.natives.annotations.ReturnType;
+import org.ballerinalang.util.exceptions.BallerinaException;
+
+/**
+ * Native function ballerina.io#hasNextTextRecord.
+ *
+ * @since 0.96.1
```
Comment (pubudu91) [inline modules/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/HasNextTextRecord.java]: Need to double check this. I think the next version is going to be 0.961

File: modules/ballerina-builtin/src/main/ballerina/ballerina/io/natives.bal [hunk 2]
```diff
@@ -83,3 +83,8 @@ public native function <ByteChannel channel> writeBytes(blob content,int startOf
 @Description { value:"Function to close a byte channel"}
 @Param {value:"channel: The ByteChannel to be closed"}
 public native function <ByteChannel channel> close();
+
+@Description {value:"Function to check whether next record available"}
```
Comment (pubudu91) [inline modules/ballerina-builtin/src/main/ballerina/ballerina/io/natives.bal]: Typo.. record is*

File: modules/ballerina-builtin/src/main/ballerina/ballerina/io/natives.bal [hunk 3]
```diff
@@ -83,3 +83,8 @@ public native function <ByteChannel channel> writeBytes(blob content,int startOf
...
+@Param {value:"channel: The TextRecordChannel to read text records from"}
+@Return {value:"Whether another record available or not"}
```
Comment (pubudu91) [inline modules/ballerina-builtin/src/main/ballerina/ballerina/io/natives.bal]: Shall we rephrase the text in the `@Return` to the format followed in `contains()` in `string`, `hasNext()` in `datatable` etc.?

File: modules/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/HasNextTextRecord.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    @Override
+    public BValue[] execute(Context context) {
+        BBoolean hasNext;
+        try {
+            BStruct channel = (BStruct) getRefArgument(context, TXT_RECORD_CHANNEL_INDEX);
+            TextRecordChannel textRecordChannel =
+                    (TextRecordChannel) channel.getNativeData(IOConstants.TXT_RECORD_CHANNEL_NAME);
+            hasNext = new BBoolean(textRecordChannel.hasNext());
+        } catch (Throwable e) {
+            String message = "Error occurred while reading text records:" + e.getMessage();
```
Comment (pubudu91) [inline modules/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/HasNextTextRecord.java]: Can we make sure that the potential throwables which end up here doesn't contain messages that,
1. Doesn't make sense without context (i.e: some exceptions such as `ArrayIndexOutOfBoundsException` contain only the invalid index as the message. The message only makes sense with the name of the exception.)
2. Doesn't expose implementation details (i.e: some exceptions contain Java exception and class names)

File: modules/ballerina-builtin/src/main/ballerina/ballerina/io/natives.bal [hunk 5]
```diff
@@ -83,3 +83,8 @@ public native function <ByteChannel channel> writeBytes(blob content,int startOf
 @Description { value:"Function to close a byte channel"}
 @Param {value:"channel: The ByteChannel to be closed"}
 public native function <ByteChannel channel> close();
+
+@Description {value:"Function to check whether next record is available or not"}
+@Param {value:"channel: The TextRecordChannel to read text records from"}
+@Return {value:"Whether another record available or not"}
```
Comment (pubudu91) [inline modules/ballerina-builtin/src/main/ballerina/ballerina/io/natives.bal]: Shall we rephrase the text in the @Return to the format followed in contains() in string, hasNext() in datatable etc.?

File: modules/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/NextTextRecord.java [hunk 6]
```diff
@@ -30,20 +30,20 @@
...
+        functionName = "nextTextRecord",
         receiver = @Receiver(type = TypeKind.STRUCT, structType = "TextRecordChannel", structPackage = "ballerina.io"),
         returnType = {@ReturnType(type = TypeKind.ARRAY, elementType = TypeKind.STRING)},
         isPublic = true
 )
-public class ReadTextRecord extends AbstractNativeFunction {
+public class NextTextRecord extends AbstractNativeFunction {
     /**
-     * Specifies the index which contains the byte channel in ballerina.lo#readTextRecord.
+     * Specifies the index which contains the byte channel in ballerina.lo#nextTextRecord.
```
Comment (anupama-pathirage) [inline modules/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/NextTextRecord.java]: Package name typo "ballerina.lo

**Comment by wggihan** [comment]
This will fix through a new PR https://github.com/ballerina-lang/ballerina/pull/4553

**Pull request closed without merging**